src/java/org/apache/cassandra/io/compress/CompressionMetadata.java

```diff
@@ -263,32 +327,28 @@ public Chunk chunkFor(long position)
     public long getTotalSizeForSections(Collection<SSTableReader.PartitionPositionBounds> sections)
     {
         long size = 0;
-        long lastOffset = -1;
         for (SSTableReader.PartitionPositionBounds section : sections)
         {
-            int startIndex = (int) (section.lowerPosition / parameters.chunkLength());
-
-            int endIndex = (int) (section.upperPosition / parameters.chunkLength());
-            if (section.upperPosition % parameters.chunkLength() == 0)
-                endIndex--;
+            int sectionStartIdx = chunkIndex(section.lowerPosition);
```

Comment:
Are these positions zero-copy-adjusted?

Reply:
`chunkIndex` returns an index for the partial file